File: 12-integer-to-roman.go
type N struct{
    Rom string
    Int int
}

func intToRoman(num int) string {
    cmap_list := [] N{
        N{"M", 1000}, N{"CM", 900}, N{"D", 500}, N{"CD", 400}, N{"C", 100},
        N{"XC", 90}, N{"L", 50}, N{"XL", 40}, N{"X", 10}, N{"IX", 9},
        N{"V", 5}, N{"IV", 4}, N{"I", 1},
    }
    s := ""
    for _, n := range cmap_list {
        for num >= n.Int {
            num -= n.Int
            s += n.Rom
        }
    }
    return s
}

File: bwu/math/12-integer-to-roman.py
class Solution(object):
    def intToRoman(self, num):
        """
        :type num: int
        :rtype: str
        """
        nums = [1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1]
        symbols = ["M", "CM", "D", "CD", "C", "XC", "L", "XL", "X", "IX", "V", "IV", "I"]
        
        res = ''
        
        if num <= 0:
            return res
            
        digit = 0
        while num > 0:
            times = num / nums[digit]
            num -= nums[digit] * times
            res += symbols[digit] * times
            digit += 1
        return res
